src/main/java/com/blamejared/crafttweaker/impl/managers/GenericRecipesManager.java

```diff
@@ -0,0 +1,177 @@
+ * You can access this manager by using the `recipes` global keyword.
+ *
+ * @docParam this recipes
+ */
+@ZenRegister
+@Document("vanilla/api/managers/GenericRecipesManager")
+@ZenCodeType.Name("crafttweaker.api.GenericRecipesManager")
+public class GenericRecipesManager {
+    
+    
+    @ZenCodeGlobals.Global("recipes")
+    public static final GenericRecipesManager recipes = new GenericRecipesManager();
+    
+    /**
+     * Add a new recipe based on the given recipe in a valid DataPack JSON format.
+     *
+     * Unlike the addJSONRecipe method in {@link IRecipeManager} you **must** set the type of the recipe within the JSON yourself.
+     * This allows you to explicitly specify a serializer if their name does not match the RecipeType name.
+     *
+     * @param name The recipe's resource path
+     * @param data The recipe in JSON format
+     * @docParam name "recipe_name"
+     * @docParam data {
+     * type: "minecraft:smoking",
+     * ingredient: <item:minecraft:gold_ore>,
+     * result: <item:minecraft:cooked_porkchop>,
+     * experience:0.35 as float,
+     * cookingtime:100
+     * }
+     */
+    @ZenCodeType.Method
+    public void addJSONRecipe(String name, IData data) {
+        if(!(data instanceof MapData)) {
+            throw new IllegalArgumentException("Json recipe's IData should be a MapData!");
+        }
+        final JsonObject recipeObject = IRecipeManager.JSON_RECIPE_GSON.fromJson(data.toJsonString(), JsonObject.class);
+        if(!recipeObject.has("type")) {
+            throw new IllegalArgumentException("Serializer type missing!");
+        }
+        
+        final ResourceLocation recipeName = new ResourceLocation(CraftTweaker.MODID, name);
+        final IRecipe<?> result = RecipeManager.deserializeRecipe(recipeName, recipeObject);
+        final RecipeManagerWrapper recipeManagerWrapper = new RecipeManagerWrapper(result.getType());
+        CraftTweakerAPI.apply(new ActionAddRecipe(recipeManagerWrapper, result, null));
+    }
+    
+    /**
+     * Removes recipes by output
+     *
+     * @param output The recipe result
+     * @docParam output <item:minecraft:iron_ingot>
+     */
+    @ZenCodeType.Method
+    public void removeRecipes(IItemStack output) {
+        CraftTweakerAPI.apply(new ActionRemoveFromWholeRegistryByOutput(output));
+    }
+    
+    /**
+     * Removes all recipes with this name.
+     * It is possible that there exist more than one recipe with the same name, if they exist in different recipe types.
+     *
+     * @param name The recipe name to remove
+     */
+    @ZenCodeType.Method
+    public void removeByName(String name) {
+        CraftTweakerAPI.apply(new ActionRemoveFromWholeRegistryByName(name));
+    }
+    
+    /**
+     * Removes all recipes from the provided mod.
+     * Chooses the recipes based on their full recipe name, not based on output item!
+     *
+     * @param modId The mod's modId
+     * @docParam modId "crafttweaker"
```

Comment:
Maybe replace this with `"minecraft"` instead, so people don't go around accidentally removing all their scripts lol